Mathlib/Data/Vector/Basic.lean

```diff
@@ -736,4 +773,39 @@ instance : IsLawfulTraversable.{u} (flip Vector n) where
 --       q(α) q(n) q(x) ih
 -- #align vector.reflect vector.reflect
 
+@[simp]
+theorem replicate_succ : replicate (n+1) val = val ::ᵥ (replicate n val) :=
+  rfl
+
+@[simp]
+theorem get_append_cons_zero : get (append (x ::ᵥ xs) ys) ⟨0, by simp⟩ = x :=
+  rfl
+
+@[simp]
+theorem get_append_cons_succ {i : Fin (n + m)} {h} :
+    get (append (x ::ᵥ xs) ys) ⟨i+1, h⟩ = get (append xs ys) i :=
+  rfl
+
+@[simp]
+theorem append_nil : (append xs nil) = xs :=
+  by cases xs; simp[append]
+
+
+@[simp]
+theorem get_map₂ (v₁ : Vector α n) (v₂ : Vector β n) (f : α → β → γ) (i : Fin n) :
+    get (map₂ f v₁ v₂) i = f (get v₁ i) (get v₂ i) := by
+  induction v₁, v₂ using inductionOn₂
+  case nil =>
+    exact Fin.elim0 i
+  case cons x xs y ys ih =>
+    rw [map₂_cons]
+    cases i using Fin.cases
+    . simp only [get_zero, head_cons]
+    . simp only [get_cons_succ, ih]
+
+theorem get_eq_val_eq {α : Type u} {n : Nat} (x x' : Vector α n) (i : Fin n)
```

Comment:
I'm not sure if we need this congruence lemma, it's also not used anywhere.  Using `Vector.eq` directly seems to be just as straightforward.

Reply:
We indeed don't, removed it